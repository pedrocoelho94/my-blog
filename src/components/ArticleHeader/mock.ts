import { data } from 'api/mockPosts.json'

const { id, title, excerpt, cover, author, categories, createdAt } =
  data.posts[0]

export default {
  id,
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt
}
