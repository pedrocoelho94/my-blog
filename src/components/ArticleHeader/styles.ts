import styled, { css } from 'styled-components'
import * as HeadingStyles from 'components/Heading/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles.Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.font.sizes.large};
    }
  `}
`

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: calc(${theme.font.sizes.medium} * 0.9);
  `}
`

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`
