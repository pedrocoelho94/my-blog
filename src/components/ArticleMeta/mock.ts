export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'pedro-coelho',
    displayName: 'Pedro Coelho'
  },
  categories: [
    {
      id: '78724ec23f',
      displayName: 'Tech',
      slug: 'tech'
    },
    {
      id: '762vg214g34',
      displayName: 'JS',
      slug: 'javascript'
    }
  ]
}
