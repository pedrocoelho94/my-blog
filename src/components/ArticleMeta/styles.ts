import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child):after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.2s ease-in-out;

      &:hover {
        filter: brightness(0.5);
      }
    }
  `}
`
