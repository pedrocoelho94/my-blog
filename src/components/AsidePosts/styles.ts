import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Content = styled.a`
  ${({ theme }) => css`
    display: flex;
    gap: 1rem;
    height: 100%;
    cursor: pointer;
    text-decoration: none;
    margin: ${theme.spacings.medium} 0;
  `}
`

export const Cover = styled.img`
  border-radius: 0.8rem;
  width: 10rem;
  height: 10rem;
  object-fit: cover;
`

export const Title = styled.h4`
  ${({ theme }) => css`
    font-size: 1.5rem;
    height: 100%;
    margin: 0;
    color: ${theme.colors.darker};

    &:hover {
      text-decoration: underline;
    }
  `}
`
