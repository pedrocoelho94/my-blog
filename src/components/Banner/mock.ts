import { data } from 'api/mockPosts.json'

export default {
  settings: data.setting,
  post: data.posts[0]
}
