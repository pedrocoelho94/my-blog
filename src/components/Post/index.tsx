import ArticleHeader, { ArticleHeaderProps } from 'components/ArticleHeader'
import HtmlContent from 'components/HtmlContent'

import * as S from './styles'

export type PostsProps = {
  content: string
} & ArticleHeaderProps

const Post = ({
  id,
  title,
  excerpt,
  cover,
  author,
  createdAt,
  categories,
  content
}: PostsProps) => (
  <S.Wrapper>
    <ArticleHeader
      id={id}
      title={title}
      excerpt={excerpt}
      cover={cover}
      author={author}
      categories={categories}
      createdAt={createdAt}
    />

    <HtmlContent html={content} />
  </S.Wrapper>
)

export default Post
