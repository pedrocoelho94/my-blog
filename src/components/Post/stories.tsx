import { Story, Meta } from '@storybook/react'
import Post, { PostsProps } from '.'

import mock from './mock'

export default {
  title: 'Posts/Post',
  component: Post,
  args: mock
} as Meta

export const Default: Story<PostsProps> = (args) => (
  <div>
    <Post {...args} />
  </div>
)
