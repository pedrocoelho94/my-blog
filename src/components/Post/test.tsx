import { screen } from '@testing-library/react'
import { renderWithTheme } from 'styles/renderWithTheme'
import { formatDate } from 'utils/format-date'

import mock from './mock'
import Post, { PostsProps } from '.'

const props: PostsProps = mock

describe('<Post />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Post {...props} />)

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()

    expect(screen.getByText(props.excerpt)).toHaveStyle({
      'font-size': '2.4rem'
    })

    expect(
      screen.getByRole('img', { name: props.cover.alternativeText })
    ).toBeInTheDocument()

    expect(screen.getByText(formatDate(props.createdAt))).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
