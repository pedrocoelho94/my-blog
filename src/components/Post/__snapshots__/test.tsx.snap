// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should render correctly 1`] = `
.c6 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c6 .categories span:not(:last-child):after {
  content: ', ';
}

.c6 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c6 a:hover {
  -webkit-filter: brightness(0.5);
  filter: brightness(0.5);
}

.c3 {
  color: #000000;
  line-height: 1.2;
  font-size: 4.0rem;
  text-transform: none;
}

.c1 {
  padding-bottom: 4.0rem;
  margin-bottom: 4.0rem;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c1 .c2 {
  margin: 0;
  margin-bottom: 2.4rem;
  font-size: 3.2rem;
}

.c4 {
  margin: 2.4rem 0;
  font-size: calc(2.4rem * 0.9);
}

.c5 {
  max-width: 100%;
  display: block;
  margin-bottom: 2.4rem;
}

.c0 {
  width: 100%;
  max-width: 120rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(3.2rem / 2);
  padding-right: calc(3.2rem / 2);
}

.c7 {
  font-size: calc(1.6rem + 0.2rem);
  line-height: 1.5;
}

.c7 p {
  margin: 4.0rem 0;
}

.c7 a,
.c7 a:visited,
.c7 a:link {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c7 a:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

.c7 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c7 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c7 pre code {
  color: inherit;
  background: inherit;
}

.c7 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c7 img {
  max-width: 100%;
}

.c7 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c7 .image figcaption {
  font-size: 1.6rem;
  padding: 0.8rem;
  text-align: center;
  line-height: 1.3;
}

.c7 ul,
.c7 ol {
  margin: 2.4rem 4.0rem;
}

.c7 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c7 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c7 table td,
.c7 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c7 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: calc(1.6rem * 1.2);
  }
}

@media (min-width:768px) {
  .c0 {
    padding-left: calc(3.2rem * 0.7);
    padding-right: calc(3.2rem * 0.7);
  }
}

@media (min-width:768px) {
  .c7 .image-style-side {
    float: right;
    max-width: 50%;
    margin: 2.4rem;
  }
}

@media (max-width:768px) {
  .c7 {
    font-size: 2rem;
  }
}

<div>
  <article
    class=""
  >
    <div
      class="c0"
    >
      <header
        class="c1"
      >
        <h1
          class="c2 c3"
        >
          Com dois pênaltis, Napoli vira contra o Legia e se recupera em seu grupo da Liga Europa
        </h1>
        <p
          class="c4"
        >
          Após levar apenas um ponto nas duas primeiras rodadas, o Napoli chegou a sete se colocou em situação mais confortável para a reta final.
        </p>
        <img
          alt="Texto Alternativo"
          class="c5"
          src="https://res.cloudinary.com/rabbitz/image/upload/v1636129861/zielinski_napoli_9359de0d9d.jpg"
        />
        <div
          class="c6"
        >
          <p>
            <span>
              Por 
            </span>
            <a
              href="/author/pedro-coelho"
            >
              Pedro Coelho
            </a>
            <span
              class="separator"
            >
               em 
            </span>
            <time
              datetime="2021-02-12"
            >
              11 de fev. de 2021
            </time>
            <span
              class="separator"
            >
               - 
            </span>
            <span
              class="categories"
            >
              <span>
                <a
                  href="/category/futebol"
                >
                  Futebol
                </a>
              </span>
            </span>
          </p>
        </div>
      </header>
      <div
        class="c7"
      >
        
  
        <p>
          Nostro 
          <strong>
            definitiones
          </strong>
           (
          <i>
            React
          </i>
          ) sed cu - 
          <u>
            Otávio Miranda
          </u>
          . Sea 
          <code>
            useEffect
          </code>
          , useState dicat fabulas 
          <a
            href="https://gist.github.com/luizomf/e7cf0c3d172801225cce694681ecb0ad"
          >
            consetetur
          </a>
           no, odio deseruisse assueverit mei eu, vis probo nonumes molestie id. Ipsum movet liberavisse an mea, diceret vituperata ne his, delenit apeirian quo ea. Imperdiet percipitur te eos. Per ludus aliquam cu. 
          <a
            href="https://gist.github.com/luizomf/e7cf0c3d172801225cce694681ecb0ad"
          >
            Eos ex tincidunt necessitatibus, timeam theophrastus mel cu
          </a>
          . La.
        </p>
        
  
  
        <pre>
          <code
            class="language-plaintext"
          >
            import { HtmlContent } from '../HtmlContent';
  import * as Styled from './styles';
  export type FooterProps = {
    footerHtml: string;
  };
  export const Footer = ({ footerHtml }: FooterProps) =&gt; {
    return (
      &lt;Styled.Container&gt;
        &lt;HtmlContent html={footerHtml} /&gt;
      &lt;/Styled.Container&gt;
    );
  };
  
          </code>
        </pre>
        
  
        <p>
          Qui eu 
          <i>
            mutat
          </i>
           exerci 
          <span
            style="background-color:hsl(60,75%,60%);"
          >
            dolores
          </span>
          , his no 
          <span
            style="color:hsl(270,75%,60%);"
          >
            repudiare
          </span>
           constituam 
          <s>
            accommodare
          </s>
          , vis ea sint 
          <u>
            tempor
          </u>
           eirmod. In mei eruditi moderatius, ex per iudico essent repudiandae. In vis vidisse menandri, an has repudiare disputationi delicatissimi. In has wisi justo postulant. Ei viderer ornatus mel, usu an partiendo tincidunt, ea eum brute ullamcorper.
        </p>
        <p>
          10
          <sup>
            2
          </sup>
          *5
          <sup>
            2
          </sup>
           - H
          <sub>
            2
          </sub>
          0
        </p>
        <p
          style="text-align:center;"
        >
          <strong>
            Center: 
          </strong>
          Abhorreant reformidans eum an, per equidem repudiandae no, ex cum elitr argumentum persequeris. In eam consul theophrastus delicatissimi, lobortis rationibus efficiendi mea ei, nam natum dolore scripserit ea. Sit partiendo eloquentiam ne. Sea no autem dolores, ridens gloriatur mel cu. Munere nullam epicuri his id, dicam definiebas deterruisset eu vim. Ea veritus deleniti usu, nam ei idque choro volumus, ex nisl aliquam bonorum eam.
        </p>
        <p>
          Fugit aliquip erroribus eu eam, an mel vidit doctus. Vis nobis veritus maluisset id, nonumy adversarium neglegentur est ut. Tation persius pri eu, modo minim fastidii mei no. Ad tota pertinax definiebas eam.
        </p>
        <p
          style="text-align:right;"
        >
          <strong>
            Right:
          </strong>
           Mel in lorem feugiat, tation dolore et eum. Id has utamur albucius molestiae. Dico justo maiestatis ei mei, labitur laboramus voluptatum nam ut. Saperet propriae nec an.
        </p>
        <p>
          Quod oblique mei ne, cetero delicatissimi sea cu, cu est harum dicit verterem. Tota forensibus ut usu, luptatum dignissim persecuti vel ex. Mea nostrud instructior cu, et referrentur theophrastus eum. Pro in iuvaret senserit. In persius explicari conceptam nec, in populo ponderum consequat vis. Gloriatur conclusionemque no qui, vitae nominati nec te, laoreet torquatos accommodare ex qui. Eu natum mediocritatem eos, cum ex alia oportere.
        </p>
        <p
          style="text-align:justify;"
        >
          <strong>
            Justify:
          </strong>
           Cu vis nonumes petentium disputationi, invidunt aliquando ex ius, ocurreret suscipiantur conclusionemque eu qui. Probo nemore ut est. Et eam dicta oratio eligendi. Quem dicta dolore vel et, at reque gloriatur definitionem sed.
        </p>
        <p>
          Nec simul ullamcorper at. Mea possit dolores persecuti te, ei quidam apeirian quo. Has quas corpora ne. Ius ad malis iusto delenit, sit ferri idque ex.
        </p>
        <blockquote>
          <p>
            Blockquote: Et sed legere rationibus, vitae evertitur ad eam. Eu amet vero officiis vis. Ad nullam pericula neglegentur sit, vix ei regione legendos argumentum. Mei an quas convenire, nobis dicunt mei ut. Pro no debet referrentur, ei est falli libris. Eu per summo salutandi hendrerit, salutatus democritum interesset nec ut. Wisi lobortis incorrupte ei mea, explicari interpretaris has et.
          </p>
        </blockquote>
        <p>
          A new line:
        </p>
        <hr />
        <p>
          Now keep going…
        </p>
        <figure
          class="image"
        >
          <a
            href="https://gist.github.com/luizomf/e7cf0c3d172801225cce694681ecb0ad"
          >
            <img
              alt="O alt text da imagem"
              src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614696630/Frozen_sunset_on_the_lake_by_Manuel_Arslanyan_9f9cd8ea10.jpg"
            />
          </a>
          <figcaption>
            Aqui está uma imagem
          </figcaption>
        </figure>
        <p>
          Munere euripidis an sed, usu ne consul soluta utamur. Bonorum equidem placerat eu cum, tantas quidam duo et. Pro soleat quaestio ne, ut nam wisi vocibus inciderint, ex his maiorum mnesarchum. Et est wisi primis, id vel odio alienum voluptatibus. Veri doctus offendit vim ea. Has nemore prompta inciderint ei, no sea expetendis dissentiet. Copiosae delicatissimi an sea, id cum agam clita nostro, tale tincidunt delicatissimi ei eos.
        </p>
        <figure
          class="image image-style-side"
        >
          <img
            alt="Uma imagem à direita"
            src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614696630/Frozen_sunset_on_the_lake_by_Manuel_Arslanyan_9f9cd8ea10.jpg"
          />
          <figcaption>
            Uma imagem à direita
          </figcaption>
        </figure>
        <p>
          Munere euripidis an sed, usu ne consul soluta utamur. Bonorum equidem placerat eu cum, tantas quidam duo et. Pro soleat quaestio ne, ut nam wisi vocibus inciderint, ex his maiorum mnesarchum. Et est wisi primis, id vel odio alienum voluptatibus. Veri doctus offendit vim ea. Has nemore prompta inciderint ei, no sea expetendis dissentiet. Copiosae delicatissimi an sea, id cum agam clita nostro, tale tincidunt delicatissimi ei eos.
        </p>
        <p>
          Munere euripidis an sed, usu ne consul soluta utamur. Bonorum equidem placerat eu cum, tantas quidam duo et. Pro soleat quaestio ne, ut nam wisi vocibus inciderint, ex his maiorum mnesarchum. Et est wisi primis, id vel odio alienum voluptatibus. Veri doctus offendit vim ea. Has nemore prompta inciderint ei, no sea expetendis dissentiet. Copiosae delicatissimi an sea, id cum agam clita nostro, tale tincidunt delicatissimi ei eos.
        </p>
        <p>
          Solum consul scripserit in duo, an sit postea denique percipit. Facer ignota consequat nec ei, idque etiam corrumpit sit ut. In mucius definitionem quo, vis an nisl diam appellantur, hendrerit comprehensam ea per. Vix saepe fierent ut.
        </p>
        <p>
          Cu mei summo copiosae. Et vim quaestio partiendo. Quot praesent imperdiet at duo. Sit probo causae ei. Ad ius nemore eruditi.
        </p>
        <p>
          Mei in qualisque disputando. Adipisci philosophia et sed, mel diceret patrioque sententiae ea, iriure posidonium consectetuer ad eam. Nobis reprimique duo in, eam corpora mandamus ex. In quod euismod eum, ne habemus cotidieque necessitatibus cum. Liber recteque duo no, ludus reprehendunt has an.
        </p>
        <p>
          Solum consul scripserit in duo, an sit postea denique percipit. Facer ignota consequat nec ei, idque etiam corrumpit sit ut. In mucius definitionem quo, vis an nisl diam appellantur, hendrerit comprehensam ea per. Vix saepe fierent ut.
        </p>
        <p>
          Cu mei summo copiosae. Et vim quaestio partiendo. Quot praesent imperdiet at duo. Sit probo causae ei. Ad ius nemore eruditi.
        </p>
        <p>
          Mei in qualisque disputando. Adipisci philosophia et sed, mel diceret patrioque sententiae ea, iriure posidonium consectetuer ad eam. Nobis reprimique duo in, eam corpora mandamus ex. In quod euismod eum, ne habemus cotidieque necessitatibus cum. Liber recteque duo no, ludus reprehendunt has an.
        </p>
        <p>
          Bullet:
        </p>
        <ul>
          <li>
            Mei in qualisque disputando. Adipisci philosophia et sed, mel diceret patrioque sententiae ea, iriure posidonium consectetuer ad eam. Nobis reprimique duo in, eam corpora mandamus ex. In quod euismod eum, ne habemus cotidieque necessitatibus cum. Liber recteque duo no, ludus reprehendunt has an.
          </li>
          <li>
            iriure posidonium consectetuer ad eam.
          </li>
          <li>
            Nobis reprimique duo in, eam corpora mandamus
          </li>
          <li>
            ex. In
          </li>
          <li>
            quod euismod
          </li>
          <li>
            eum, ne habemus cotidieque necessitatibus cum.
          </li>
          <li>
            Liber recteque duo no, ludus reprehendunt has an.
          </li>
        </ul>
        <p>
          Now ordered:
        </p>
        <ol>
          <li>
            Item 1
          </li>
          <li>
            Item
          </li>
          <li>
            Item
          </li>
          <li>
            Item
          </li>
        </ol>
        <p>
          A think that's it…
        </p>
        <figure
          class="table"
        >
          <table>
            <thead>
              <tr>
                <th>
                  Title 1
                </th>
                <th>
                  Title 1
                </th>
                <th>
                  Title 1
                </th>
                <th>
                  Title 1
                </th>
                <th>
                  Title 1
                </th>
                <th>
                  Title 1
                </th>
                <th>
                  Title 1
                </th>
                <th>
                  Title 1
                </th>
                <th>
                  Title 1
                </th>
                <th>
                  Title 1
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
              </tr>
              <tr>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
              </tr>
              <tr>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
                <td>
                  Item
                </td>
              </tr>
            </tbody>
          </table>
        </figure>
        <p>
          A video:
        </p>
        <figure
          class="media"
        >
          <oembed
            url="https://www.youtube.com/watch?v=oLmRV25eQvo"
          />
        </figure>
        <p>
          The end.
        </p>
        
  
      </div>
    </div>
  </article>
</div>
`
